components/renku_data_services/notebooks/api/amalthea_patches/init_containers.py

```diff
@@ -189,35 +332,28 @@ def certificates(config: _NotebooksConfig) -> list[dict[str, Any]]:
                 {
                     "op": "add",
                     "path": "/statefulset/spec/template/spec/initContainers/-",
-                    "value": api_client.sanitize_for_serialization(init_container),
-                },
-            ],
-        },
-        {
-            "type": "application/json-patch+json",
-            "patch": [
-                {
-                    "op": "add",
-                    "path": "/statefulset/spec/template/spec/volumes/-",
-                    "value": api_client.sanitize_for_serialization(volume_etc_certs),
-                },
-            ],
-        },
-        {
-            "type": "application/json-patch+json",
-            "patch": [
-                {
-                    "op": "add",
-                    "path": "/statefulset/spec/template/spec/volumes/-",
-                    "value": api_client.sanitize_for_serialization(volume_custom_certs),
+                    "value": api_client.sanitize_for_serialization(container),
                 },
             ],
         },
     ]
+    for vol in vols:
+        patches.append(
+            {
+                "type": "application/json-patch+json",
+                "patch": [
+                    {
+                        "op": "add",
+                        "path": "/statefulset/spec/template/spec/volumes/-",
+                        "value": api_client.sanitize_for_serialization(vol),
+                    },
+                ],
+            },
+        )
     return patches
 
 
-def download_image(server: "UserServer") -> list[dict[str, Any]]:
+def download_image_container(server: "UserServer") -> client.V1Container:
```

Comment:
Just to be sure: it's an image puller ?

Reply:
Yes this is a container used in the old code that just runs first (or among the first) and just downloads the image. The status updates for sessions (in v1 for the old amalthea) are tied to each container in the session. So having such a container allows us to surface in our status how long it takes to just download the session image. This is not used with the new amalthea btw.